spec/app/ics-029-fee-payment/README.md

```diff
@@ -0,0 +1,223 @@
+[general fee payment design](https://github.com/cosmos/ibc/issues/578) that could be adopted by any ICS application
+protocol.
+
+In general, the Interchain dream will never scale unless there is a clear way to incentivize relayers. We seek to
+define a clear interface that can be easily adopted by any application, but not preclude chains that don't use tokens.
+
+### Desired Properties
+
+- Incentivize timely delivery of the packet (`OnReceivePacket` called)
+- Incentivize relaying acks for these packets (`OnAcknowledgement` called)
+- Incentivize relaying timeouts for these packets when the timeout has expired before packet is delivered (for example as receive fee was too low) (`OnTimeout` called)
+- Produces no extra IBC packets
+- One direction works, even when one chain does not support concept of fungible tokens
+- Opt-in for each chain implementing this. eg. ICS27 with fee support on chain A could connect to ICS27 without fee support on chain B.
+- Standardized interface for each chain implementing this extension
+- Support custom fee-handling logic within the same framework
+- Relayer addresses should not be forgable
+- Permissionless relaying
+
+## Technical Specification
+
+### General Design
+
+In order to avoid extra packets on the order of the number of fee packets, as well as provide an opt-in approach, we
+store all fee payment info only on the source chain. The source chain is the one location where the sender can provide tokens
+to incentivize the packet. The fee distribution may be implementation specific and thus does not need to be in the ibc spec
+(just high-level requirements are needed in this doc).
+
+We require that the [relayer address is exposed to application modules](https://github.com/cosmos/ibc/pull/579) for
+all packet-related messages, so the modules are able to incentivize the packet relayer. `OnAcknowledgement`, `OnTimeout`,
+and `OnTimeoutClose` messages will therefore have the relayer address and be capable of sending escrowed tokens to such address.
+However, we need a way to reliably get the address of the relayer that submitted `OnReceivePacket` on the destination chain to
+the source chain. In fact, we need a *source address* for this relayer to pay out to, not the *destination address* that signed
+the packet.
+
+Given this, the flow would be:
+
+1. User/module submits a send packet on the `source` chain, along with some tokens and fee information on how to distribute them. The fee tokens are all escrowed by the fee module.
+2. Relayer submits `OnReceivePacket` on the `destination` chain. Along with this message, the *forward relayer* will submit a `payToOnSource` address where payment should be sent.
+3. Destination application includes this `payToOnSource` in the acknowledgement (there are multiple approaches to discuss below)
+4. Relayer submits `OnAcknowledgement` which provides the *return relayer* address on the source chain, along with the `payToOnSource` address
```

Comment:
This is _not necessarily the same relayer_, right? We should clarify (e.g. name them relayers 1, 2 or A, B)